### DISCUSSION TITLE
Is it possible to use Phlex as templating language?

### DISCUSSION THREAD

**Discussion by Urist-McUristurister** [yippee-fun/phlex#589] [Help]
Is it possible to use Phlex as templating language, akin to Markaby or Arbre? Without all the hassle of an extra class, all the methods, variables, etc. Just a file, say, `app/views/posts/index.html.phlex`, that would contain all the code that normally goes into `template` method of a Phlex component, and let Rails (or, for example, Tilt) take care of everything else.

If it's not possible at the moment, are there, maybe, any plans to enable this in the future?..

Thanks!

**Comment by joelmoss** [comment]
Ooh, I kinda like that idea!

So simply define a template `app/views/posts/index.html.phlex` with contents like this:

```ruby
h1 do
 "hello"
end
```

Which is essentially just the content of the `template` method, and have Rails render it as it would any other template, with access to controller instance variables, etc.

It's definately not something you would want to do all the time, but as a conveniant way to create quick and simple views, it really nice.

**Reply by Urist-McUristurister** [reply]
Yeah, that's exactly what I mean! ;-)
